### ISSUE TITLE
ref first table get a  WARNING: undefined label

### ISSUE THREAD

**Issue by WarlockFish** [sphinx-doc/sphinx#9638]
### Describe the bug

when I write a numref for a table , i get a Warnning (all table is warnning , list-table 、table......)

```
make latexpdf

rst:433: WARNING: undefined label: t2-1
```

```
如 :numref:`t2-1` 是对这三个 Raid 级别的比较。
```

```
.. _t2-1:
.. list-table:: Raid 级别比较
   :widths: 20 30 30 30
   :header-rows: 1

   * - Raid 级别
     - Raid 0
     - Raid 1
     - Raid 5
   * - 别名
     - 条带
     - 镜像 
     - 分布奇偶位条带

```

but when i use other way ,it can pass

```
.. list-table:: Raid 级别比较
   :widths: 20 30 30 30
   :header-rows: 1
   :name: t2-1

   * - Raid 级别
     - Raid 0
     - Raid 1
     - Raid 5
   * - 别名
     - 条带
     - 镜像 
     - 分布奇偶位条带
```

and the sphix 4.1.2 and 4.2 can get this error ,the sphinx 3.5.4  is pass 

### How to Reproduce

```
$ git clone https://github.com/WarlockFish/spinx-error.git
$ cd spinx-error
$ pip install -r requirements.txt
$ make latexpdf
$ # open build/latex/xxx.pdf
```

### Your project

https://github.com/WarlockFish/spinx-error.git

### OS

linux

### Python version

3.8

### Sphinx version

4.2 4.1.6  3.5.4

**Comment by tk0miya** [comment]
Note: I reproduced the errow with this Dockefile:
```
FROM python:3.8-slim

RUN apt update; apt install -y build-essential curl git unzip vim
RUN git clone https://github.com/WarlockFish/spinx-error.git
WORKDIR spinx-error
RUN apt install -y libxml2-dev libxslt-dev
RUN sed -i -e 's/docutils==0.17.1/docutils==0.16/' requirements.txt
RUN apt install -y libx11-dev
RUN apt install -y libxext-dev
RUN pip install -r requirements.txt
RUN make latex
```

**Comment by tk0miya** [comment]
I found this is a bug of docutils (a core library of Sphinx).

It will build document wrongly:

Input:
```
A test document to check the behavior of docutils.

.. _table1:
.. list-table:: Table with hyperlink target

   * - header

.. list-table:: Table with name
   :name: table2

   * - header

A reference to `table1`_ will not work because generated HTML does not contain `#table1` element if `--language` option is set to non-English.
A reference to `table2`_ will work even if any `--language` option given.
```

Command:
```
$ rst2html.py test.rst -l zh_cn
```

Result:
```
<div class="document">


<p>A test document to check the behavior of docutils.</p>
<table border="1" class="docutils">
<caption>Table with hyperlink target</caption>
<colgroup>
<col width="100%" />
</colgroup>
<tbody valign="top">
<tr><td>header</td>
</tr>
</tbody>
</table>
<table border="1" class="docutils" id="table2">
<caption>Table with name</caption>
<colgroup>
<col width="100%" />
</colgroup>
<tbody valign="top">
<tr><td>header</td>
</tr>
</tbody>
</table>
<p>A reference to <a class="reference internal" href="#table1">table1</a> will not work because generated HTML does not contain <cite>#table1</cite> element if <cite>--language</cite> option is set to non-English.
A reference to <a class="reference internal" href="#table2">table2</a> will work even if any <cite>--language</cite> option given.</p>
</div>
```

**Comment by tk0miya** [comment]
I reported this as https://sourceforge.net/p/docutils/bugs/425/.

**Comment by gmilde** [comment]
> I reported this as https://sourceforge.net/p/docutils/bugs/425/.

Fixed in Docutils 0.18. Now we only need Sphinx support for this version (cf. #9777).

**Comment by AA-Turner** [comment]
Sphinx now supports Docutils 0.18 and 0.19.

A
